packages/dds/matrix/src/matrix.ts

```diff
@@ -357,16 +357,22 @@ export class SharedMatrix<T extends Serializable = Serializable>
     /** @internal */ public _undoRemoveRows(segment: ISegment) {
         const original = segment as PermutationSegment;
 
-        // (Re)insert the removed number of columns at the original position.
+        // (Re)insert the removed number of rows at the original position.
         const rowStart = this.rows.getPosition(original);
         this.insertRows(rowStart, original.cachedLength);
 
-        // Transfer handles from the original segment to the newly inserted segment.
-        // (This allows us to use getCell(..) below to read the previous cell values)
+        // Transfer handles from the original segment to the newly inserted empty segment.
```

Comment:
The part of the comment in parenthesis was stale: we now read directly from the tiled array.